src/gt4py/next/ffront/past_to_itir.py

```diff
@@ -263,7 +273,10 @@ def _construct_itir_initialized_domain_arg(
                 f"expected '{dim}', got '{keys_dims_types}'."
             )
 
-        return [self.visit(bound) for bound in node_domain.values_[dim_i].elts]
+        itir_node = [self.visit(bound) for bound in node_domain.values_[dim_i].elts]
```

Comment:
Why is not taken care of by the `PreserveLocationWithSymbolTableTrait`? Aren't you just manually copying the location of the bound you are also visiting?